packages/shared/src/components/modals/OnboardingModal.tsx

```diff
@@ -85,82 +75,20 @@ function OnboardingModal({
     return setIsClosing(true);
   };
 
-  const onBackStep = (
-    beforeStep: number,
-    stepNow: number,
-    e: MouseEvent | KeyboardEvent,
-  ) => {
-    if (beforeStep === 0 || stepNow === 0) {
-      onClose(e);
-      return true;
-    }
-
-    const screen = getScreen(onboardingSteps, step);
-    trackEvent({
-      event_name: AnalyticsEvent.ClickOnboardingBack,
-      extra: JSON.stringify({ screen_value: screen }),
-    });
-    return setStep(stepNow);
-  };
-
-  const onNextStep = (beforeStep: number, stepNow: number) => {
-    const screen = getScreen(onboardingSteps, beforeStep);
-    trackEvent({
-      event_name: AnalyticsEvent.ClickOnboardingNext,
-      extra: JSON.stringify({ screen_value: screen }),
-    });
-    setStep(stepNow);
-  };
+  const { onContainerChange, formRef } = useAuthForms({ onDiscard: onClose });
 
-  const onSelectedTopicsChange = (result: Record<string, boolean>) => {
-    topics.current = result;
-    if (invalidMessage) {
-      setInvalidMessage(null);
-    }
-  };
   const components: Record<OnboardingStep, ReactNode> = {
-    topics: (
-      <FilterOnboarding
-        key={OnboardingStep.Topics}
-        preselected={topics?.current}
-        onSelectedChange={onSelectedTopicsChange}
-      />
-    ),
-    layout: <LayoutOnboarding key={OnboardingStep.Layout} />,
-    theme: <ThemeOnboarding key={OnboardingStep.Theme} />,
+    intro: null,
+    topics: <FilterOnboarding key={OnboardingStep.Topics} onClose={onClose} />,
+    layout: <LayoutOnboarding key={OnboardingStep.Layout} onClose={onClose} />,
+    theme: <ThemeOnboarding key={OnboardingStep.Theme} onClose={onClose} />,
   };
 
-  const onValidateFilter = () => {
-    const selected = Object.values(topics.current).filter((value) => !!value);
-    const isValid = selected.length >= onboardingMinimumTopics;
-    const topic = `topic${onboardingMinimumTopics > 1 ? 's' : ''}`;
-    const errorMessage = `Choose at least ${onboardingMinimumTopics} ${topic} to follow`;
-    setInvalidMessage(isValid ? null : errorMessage);
-
-    return isValid;
-  };
-
-  // each step can have their own validation before moving forward with the steps.
-  // an array index represents which step it actually is.
-  // the modal will check if the current step has its relevant validation.
-  const nextButtonValidations = useMemo(() => {
-    const length = OnboardingStep.Topics.length + INTRODUCTION_ADDITIONAL_STEP;
-    const validations = Array(length).fill(null);
-    const filterStep = onboardingSteps?.indexOf(OnboardingStep.Topics);
-
-    if (filterStep !== -1) {
-      const index = filterStep + INTRODUCTION_ADDITIONAL_STEP;
-      validations[index] = onValidateFilter;
-    }
-
-    return validations;
-  }, [onboardingSteps]);
-
   useEffect(() => {
     trackEvent({
       event_name: AnalyticsEvent.Impression,
       target_type: TargetType.MyFeedModal,
-      target_id: OnboardingVersion.V2,
+      target_id: 'v2',
```

Comment:
This one doesn't seem right?

Reply:
Oh noticed you removed the flag :)

Reply:
Yes, I also did some cleanup since the onboarding versions are now concluded.